addons/l10n_pt_account/tests/test_l10n_pt_account.py

```diff
@@ -0,0 +1,161 @@
+# -*- coding: utf-8 -*-
+from unittest.mock import patch
+from odoo import fields, _, Command
+from odoo.addons.account.tests.common import AccountTestInvoicingCommon
+from odoo.exceptions import UserError
+from odoo.models import Model
+from odoo.tests import tagged
+from odoo.tools import format_date
+import datetime
+
+
+@tagged('post_install_l10n', 'post_install', '-at_install')
+class TestL10nPtAccount(AccountTestInvoicingCommon):
+    @classmethod
+    def setUpClass(cls, chart_template_ref='l10n_pt_account.pt_chart_template'):
+        super().setUpClass(chart_template_ref=chart_template_ref)
+        cls.company_data['company'].write({
+            'street': '250 Executive Park Blvd, Suite 3400',
+            'city': 'Lisboa',
+            'zip': '9415-343',
+            'company_registry': '123456',
+            'phone': '+351 11 11 11 11',
+            'country_id': cls.env.ref('base.pt').id,
+            'vat': 'PT123456789',
+        })
+        cls.company_pt = cls.company_data['company']
+        cls.company_data['default_journal_sale'].restrict_mode_hash_table = True
+        cls.company_data['default_journal_purchase'].restrict_mode_hash_table = True
+
+    @classmethod
+    def create_invoice(cls, move_type, invoice_date="2022-01-01", create_date=None, amount=1000.0, post=False):
+        move = cls.env['account.move'].create({
+            'move_type': move_type,
+            'partner_id': cls.partner_a.id,
+            'invoice_date': fields.Date.from_string(invoice_date),
+            'line_ids': [
+                Command.create({
+                    'name': 'Product A',
+                    'quantity': 1,
+                    'price_unit': amount,
+                    'tax_ids': [],
+                }),
+            ],
+        })
+        # Bypass ORM to update the create_date
+        move._cr.execute('''
+            UPDATE account_move
+               SET create_date = %s
+             WHERE id = %s
+        ''', (datetime.datetime.strptime(create_date, '%Y-%m-%dT%H:%M:%S') if create_date else invoice_date, move.id))
+        move.invalidate_model(['create_date'])
+        if post:
+            move.action_post()
+        return move
+
+    def test_l10n_pt_account_hash_sequence(self):
+        """
+        Test that the hash sequence is correct.
+        For this, we use the following resource provided by the Portuguese tax authority:
+        https://info.portaldasfinancas.gov.pt/apps/saft-pt01/local/saft_idemo599999999.xml
+        We create invoices with the same info as in the link, and we check that the hash that we obtain in Odoo
+        is the same as the one given in the link (using the same sample keys).
+        """
+
+        # This patch is necessary because we use the move_type in l10n_pt_document_number, but our
+        # move types (out_invoice, out_refund, ...) are different from the ones used in the link (FT, NC, ...)
```

Comment:
In this case I would need to patch how the `move_type` is retrieved to map it which would still need a patch ?
Because these `move_type`s are only for the tests. In practice, they can be anything we want so we shouldn't map it in the `compute_l10n_pt_document_number` either